File: Backend/WebAPI/Src/Core/Entities/HistoricalPeriod.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Entities;

public class HistoricalPeriod
{
    public Guid Id { get; }

    [Column(TypeName = "VARCHAR(1024)")]
    public required string Name { get; set; }

    public required uint StartYear { get; set; }
    public required uint EndYear { get; set; }
}

File: Backend/WebAPI/src/main/java/com/vhl/webapi/entities/specific/HistoricalPeriod.java
package com.vhl.webapi.entities.specific;

import com.vhl.webapi.entities.superclass.IBaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Data
@Table(name = "historical_periods")
public class HistoricalPeriod extends IBaseEntity {
    @Column(name = "name", nullable = false, length = 1024)
    private String name;

    @Column(name = "start_year", nullable = false)
    private int startYear;

    @Column(name = "end_year", nullable = false)
    private int endYear;

}
